src/Common/src/System/IO/PathInternal.Unix.cs

```diff
@@ -41,5 +41,22 @@ internal static bool IsDirectorySeparator(char c)
             Debug.Assert(Path.DirectorySeparatorChar == Path.AltDirectorySeparatorChar);
             return c == Path.DirectorySeparatorChar;
         }
+
+
+        /// <summary>
+        /// Returns true if the path is too long
+        /// </summary>
+        internal static bool IsPathTooLong(string fullPath)
+        {
+            return fullPath.Length >= Interop.libc.MaxPath;
```

Comment:
I see from reading further in the changes that we were already doing this comparison, so this isn't new.  Is it correct though, on both Unix and Windows?